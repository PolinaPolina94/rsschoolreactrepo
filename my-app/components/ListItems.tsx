import { Item } from '@/types/types';
import ItemOnPage from './ItemOnPage';

type Person = {
  items: Item[];
};

const ListItems = (props: Person) => {
  const items = props.items;
  if (items && items.length) {
    return (
      <>
        {items.map((el: Item) => (
          <ItemOnPage
            key={el.id}
            name={el.name}
            image={el.image}
            species={el.species}
            type={el.type}
            gender={el.gender}
            planet={el.origin!.name}
            id={el.id}
          />
        ))}
      </>
    );
  } else {
    return <div> no such person </div>;
  }
};

export default ListItems;
