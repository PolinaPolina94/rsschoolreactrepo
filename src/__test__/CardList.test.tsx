import { screen, render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import ListItems from '../components/main-page/ListItems';

jest.mock('react-redux');

describe('CardList tests', () => {
  const objectState = {
    items: [
      {
        id: 1,
        name: 'Johnny Depp',
        image: 'https://rickandmortyapi.com/api/character/avatar/183.jpeg',
        status: 'Alive',
        species: 'Human',
        type: '',
        gender: 'Male',
        planet: 'Earth',
        created: '2017-12-29T18:51:29.693Z',
        origin: {
          name: 'Earth (C-500A)',
          url: 'https://rickandmortyapi.com/api/location/23',
        },
        location: {
          name: 'Earth (C-500A)',
          url: 'https://rickandmortyapi.com/api/location/23',
        },
      },
      {
        id: 183,
        name: 'Luiza',
        image: 'https://rickandmortyapi.com/api/character/avatar/183.jpeg',
        status: 'Alive',
        species: 'Human',
        type: '',
        gender: 'Male',
        planet: 'Earth',
        created: '2017-12-29T18:51:29.693Z',
        origin: {
          name: 'Earth (C-500A)',
          url: 'https://rickandmortyapi.com/api/location/23',
        },
        location: {
          name: 'Earth (C-500A)',
          url: 'https://rickandmortyapi.com/api/location/23',
        },
      },
    ],
  };
  it('render 2 cards in the List', async () => {
    render(
      <BrowserRouter>
        <ListItems items={objectState.items} />;
      </BrowserRouter>
    );
    expect(screen.getByText(/Johnny Depp/)).toBeInTheDocument();
    expect(screen.getByText(/Luiza/)).toBeInTheDocument();
  });
  it('show `no such persons` without cards', () => {
    render(
      <BrowserRouter>
        <ListItems items={[]} />;
      </BrowserRouter>
    );
    expect(screen.getByText(/no such person/)).toBeInTheDocument();
  });
});
