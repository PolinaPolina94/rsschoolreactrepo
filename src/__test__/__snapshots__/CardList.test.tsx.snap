// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for the ListItems component no such persons with no cards 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
         
        no such persons
         
      </div>
      ;
    </div>
  </body>,
  "container": <div>
    <div>
       
      no such persons
       
    </div>
    ;
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Tests for the ListItems component returns no cards found on wrong search 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      ;
    </div>
  </body>,
  "container": <div>
    ;
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Tests for the ListItems component show items 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="itemwrapper"
      >
        <div
          class="view"
        >
          <span>
             
            Johnny Depp
             
          </span>
          <a
            class="link"
            href="/details/1"
          >
            <img
              alt="person name"
              class="itemimg"
              id="rockNew"
              src="https://rickandmortyapi.com/api/character/avatar/183.jpeg"
            />
          </a>
        </div>
        <div
          class="description"
        >
          <span>
             Description üëæ 
          </span>
          <div
            class="descriptionList"
          >
            <span>
              species: 
              Human
            </span>
            <span>
              planet: 
              Earth (C-500A)
               ü™ê
            </span>
            <span>
              gender: 
              Male
               
            </span>
          </div>
        </div>
      </div>
      <div
        class="itemwrapper"
      >
        <div
          class="view"
        >
          <span>
             
            Johnny Depp
             
          </span>
          <a
            class="link"
            href="/details/183"
          >
            <img
              alt="person name"
              class="itemimg"
              id="rockNew"
              src="https://rickandmortyapi.com/api/character/avatar/183.jpeg"
            />
          </a>
        </div>
        <div
          class="description"
        >
          <span>
             Description üëæ 
          </span>
          <div
            class="descriptionList"
          >
            <span>
              species: 
              Human
            </span>
            <span>
              planet: 
              Earth (C-500A)
               ü™ê
            </span>
            <span>
              gender: 
              Male
               
            </span>
          </div>
        </div>
      </div>
      ;
    </div>
  </body>,
  "container": <div>
    <div
      class="itemwrapper"
    >
      <div
        class="view"
      >
        <span>
           
          Johnny Depp
           
        </span>
        <a
          class="link"
          href="/details/1"
        >
          <img
            alt="person name"
            class="itemimg"
            id="rockNew"
            src="https://rickandmortyapi.com/api/character/avatar/183.jpeg"
          />
        </a>
      </div>
      <div
        class="description"
      >
        <span>
           Description üëæ 
        </span>
        <div
          class="descriptionList"
        >
          <span>
            species: 
            Human
          </span>
          <span>
            planet: 
            Earth (C-500A)
             ü™ê
          </span>
          <span>
            gender: 
            Male
             
          </span>
        </div>
      </div>
    </div>
    <div
      class="itemwrapper"
    >
      <div
        class="view"
      >
        <span>
           
          Johnny Depp
           
        </span>
        <a
          class="link"
          href="/details/183"
        >
          <img
            alt="person name"
            class="itemimg"
            id="rockNew"
            src="https://rickandmortyapi.com/api/character/avatar/183.jpeg"
          />
        </a>
      </div>
      <div
        class="description"
      >
        <span>
           Description üëæ 
        </span>
        <div
          class="descriptionList"
        >
          <span>
            species: 
            Human
          </span>
          <span>
            planet: 
            Earth (C-500A)
             ü™ê
          </span>
          <span>
            gender: 
            Male
             
          </span>
        </div>
      </div>
    </div>
    ;
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
